runtime/Python2/src/antlr4/Recognizer.py

```diff
@@ -68,7 +74,7 @@ def getRuleIndexMap(self):
             raise UnsupportedOperationException("The current recognizer does not provide a list of rule names.")
         result = self.ruleIndexMapCache.get(ruleNames, None)
         if result is None:
-            result = zip( ruleNames, range(0, len(ruleNames)))
+            result = dict(zip(ruleNames, range(0, len(ruleNames))))
```

Comment:
According to your submission, this is not required in Python 2 ?

Reply:
The function name does, however, imply that it is returning a map. Given that we're caching it, it's probably better to cache the map rather than the list of pairs?